### PR TITLE
Revise sb56 / subst

### PR DESCRIPTION
**Pull request by benjub**
Much easier review by commit.

First commit:

The comment of sb56 mentionned it did not depend on df-sb, but the current proof did.  I fixed that (the proof is slightly longer but does not use df-sb anymore and is more direct (uses more basic theorems).  I slightly revised the comment.  I renamed it "subst", since the "56" in sb56 was a bit arbitrary (probably coming from the fact that it resulted from the combination of sb5 and sb6, both arbitrary numbered too).

There is a weakening bj-substw that is for the moment in my mathbox.

Second commit: spelling and formatting consistency.

### CODE REVIEW

File: set.mm [hunk 1]
```diff
@@ -19494,30 +19515,38 @@ Converse of the inference rule of (universal) generalization ~ ax-gen .
...
+       Wolf Lammen, 4-Sep-2023.) $)
     sb5 $p |- ( [ y / x ] ph <-> E. x ( x = y /\ ph ) ) $=
+      ( wsb weq wi wal wa wex sb6 subst bitr4i ) ABCDBCEZAFBGMAHBIABCJABCKL $.
+
+    $( Obsolete version of ~ sb5 as of 21-Sep-2024.  (Contributed by NM,
+       18-Aug-1993.)  (Proof modification is discouraged.)
+       (New usage is discouraged.) $)
+    sb5OLD $p |- ( [ y / x ] ph <-> E. x ( x = y /\ ph ) ) $=
       ( weq wa wex wsb nfs1v sbequ12 equsexv bicomi ) BCDAEBFABCGZALBCABCHABCIJ
       K $.
```
Comment (icecream17) [inline set.mm:19528]: not actually a shortening

```
ABCDBCEZAFBGMAHBIABCJABCKL
BCDAEBFABCGZALBCABCHABCIJK
```

**Comment by icecream17** [comment]
deferring until @wlammen comments (see #3458); not sure if sb56 not using df-sb is important

File: set.mm [hunk 2]
```diff
@@ -19216,6 +19215,29 @@ Converse of the inference rule of (universal) generalization ~ ax-gen .
       ( wal spimfv alrimiv ) ACGBDABCDEFHI $.
   $}
 
+  ${
```
Comment (wlammen) [inline set.mm:19218]: This points out that substitution has two implementation variants.  I wonder whether it is best placed before df-sb then.

I'd prefer equs5v, or maybe equsb45v, over subst as a name

**Review by wlammen — CHANGES_REQUESTED** [review]
Nice result.  One should come up with a better name for subst, though,  in my eyes.

**Comment by benjub** [comment]
Since ~subst shows two ways of expressing proper **subst**itution, I labeled it "subst".  Now, I think I should label it "substb", as common for **b**iconditionals, and save "subst" for the forward implication, which is equivalent to the axiom ~ax-12 (over core axioms), as ~bj-substax12 shows.  This equivalence also explains why it should be placed close after ~ax-12.

It is less clear what the label "equs(4)5v" conveys (the "equ" is for "**equ**al", then "s" maybe for "**s**ubstitution"?, then "4" and "5" because of an arbitrary numbering which has now changed, and "v" for "D**V** condition", but the statement without DV condition is not a theorem, so that prefix is not necessary, or is even unusual).

**Comment by icecream17** [comment]
maybe `sbb` would work?

**Comment by wlammen** [comment]
I don't consider equsxx a nice name, it just happens to be related to similar named existing theorems (equsal, equsex).  subst is too generic imo, substitution is a general process, but this theorem says that two implementation variants won't make much of a difference.  What about (sb/subst/equs)alex then?
I agree that it should be placed near ax-12.

**Comment by benjub** [comment]
> maybe `sbb` would work?

Label `sb*` are for theorems that involve the substitution  symbol ` [ ? / ? ] ` itself.

Another question could be: what's wrong with `substb` ?  It may be a bit long, with 6 letters, but it's not used as much as, e.g., `syl`.

**Comment by icecream17** [comment]
With sb56, you can actually predict the content of the theorem. Though sb56 made slightly more sense with the OLD proof (which is shorter); it's possible to just keep that version and delete the contradictory part of the comment. But this is a bikeshed, so this isn't a very strong opinion.

**Comment by benjub** [comment]
> What about (sb/subst/equs)alex then?

`sbalex` it will be. The point is that it's really a candidate replacement for ax-12, as shown in bj-substax12, so it's really "the axiom for substitution", and it's in that sense that it deserves to be the one with the label `subst`, IMO.

**Review by wlammen — APPROVED** [review]

**Comment by wlammen** [comment]
sbalex has a symmetric form that reminds of tertium non datur
( A. x ( x = y -> ph ) \\/ A. x ( x = y -> -. ph ) )

**Comment by benjub** [comment]
> sbalex has a symmetric form that reminds of tertium non datur ( A. x ( x = y -> ph ) \/ A. x ( x = y -> -. ph ) )

Indeed. There is also a form akin to the principle of non-contradiction, see the comment of https://us.metamath.org/mpeuni/bj-substax12.html

**Pull request merged**